device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1592,8 +1602,13 @@ private void addToCallbackQueue(IotHubTransportPacket packet)
 
     private DeviceClientConfig getDefaultConfig()
     {
-        Iterator<DeviceClientConfig> configsIterator = this.deviceClientConfigs.values().iterator();
-        return configsIterator.hasNext() ? configsIterator.next() : null;
+        for (DeviceClientConfig config : this.deviceClientConfigs.values())
+        {
+            return config;
```

Comment:
deliberately returning the first config. This is just a bit easier to read, I believe